src/edu/washington/escience/myria/accessmethod/ConnectionInfo.java

```diff
@@ -102,7 +103,13 @@ public static String toJson(final String dbms, final String hostName, final Stri
         // Now it is hardcoded to use a specific connection info, which allows only one
         // myria instance per machine in the cluster
         host = hostName;
-        port = MyriaConstants.STORAGE_MONETDB_PORT;
+            
+        if (databasePort == null) {
+            port = MyriaConstants.STORAGE_MONETDB_PORT;
+        } else {
+            port = databasePort;
```

Comment:
Use the [`firstNonNull` function](http://docs.guava-libraries.googlecode.com/git/javadoc/com/google/common/base/MoreObjects.html#firstNonNull(T,%20T)) to simplify this code:

`port = MoreObjects.firstNonNull(databasePort, MyriaConstants.STORAGE_MONETDB_PORT);`